Online/State/BodyChunkRef.cs

```diff
@@ -20,7 +20,16 @@ public BodyChunkRef(OnlinePhysicalObject owner, int index)
         {
             if (bodyChunk is null) return null;
             if (!OnlinePhysicalObject.map.TryGetValue(bodyChunk.owner.abstractPhysicalObject, out var oe))
-                throw new InvalidProgrammerException("body chunk owner doesn't exist in online space! " + bodyChunk.owner.abstractPhysicalObject);
+            {
+                for (int num = 0; num < bodyChunk.owner.abstractPhysicalObject.stuckObjects.Count; num++)
+                {
+                    if (bodyChunk.owner.abstractPhysicalObject.stuckObjects[num] is AbstractPhysicalObject.AbstractSpearStick && bodyChunk.owner.abstractPhysicalObject.stuckObjects[num].A.type == AbstractPhysicalObject.AbstractObjectType.Spear && bodyChunk.owner.abstractPhysicalObject.stuckObjects[num].A.realizedObject != null)
+                    {
+                        (bodyChunk.owner.abstractPhysicalObject.stuckObjects[num].A.realizedObject as Spear).ChangeMode(Weapon.Mode.Free);
```

Comment:
lillypuck can be stuck as well, consider casting as Weapon instead of Spear and removing the above objecttype checks

also consider using linq instead, performance impact is negligible (this isn't a hot loop)